src/utils.py

```diff
@@ -64,7 +64,7 @@ def retrieve_worker_result(external_request_id, worker):
         query = session.query(WorkerResult) \
             .filter(WorkerResult.external_request_id == external_request_id,
                     WorkerResult.worker == worker)
-        result = query.first()
+        result = query.order_by(desc(WorkerResult.ended_at)).first()
```

Comment:
@abs51295 LGTM with a minor nitpick. Please try out `WorkerResult.ended_at.desc()` instead of `desc(WorkerResult.ended_at)` to avoid an import.

For better readability, please move the `.order_by` clause to the query string in the previous line.